Dockerfile

```diff
@@ -15,12 +15,26 @@ RUN cargo build --release --locked --no-default-features
 RUN rm -f target/release/deps/git_cliff*
 
 FROM debian:buster-slim as runner
+
+# Everything inside this container will be explicitly mounted by the end user,
+# so we can sidestep some Git security restrictions. This app recommends
+# mounting data to /app, but this *can* be changed externally and *will* be
+# changed when run by GitHub Actions, so we need to cover our bases.
+RUN echo '[safe]\n\tdirectory = *' > /etc/gitconfig
+
 COPY --from=builder /app/target/release/git-cliff /usr/local/bin
-WORKDIR git-home
-RUN cat <<'EOF' > entrypoint.sh
+WORKDIR app
+
+# Even if the repository as marked as safe, GitHub Actions and some other
+# environments insist on running the entrypoint as root inside the container
+# even when being run by a non priviledged user on their own files. Here we
+# check the ownership of the workdir (which may or may not be /app) and change
+# our effective user/group ID to match.
+RUN cat <<'EOF' > /usr/local/bin/entrypoint.sh
 #!/bin/sh
-cp -r /app /git-home/app
-cd /git-home/app
-exec git-cliff "$@"
+if [ "$(id -u)" -ne "$(stat -c '%u' .)" ]; then
```

Comment:
Do you think there is any side effects of this for the `git-cliff-action` or similar projects who depend on the Docker image?

See: https://github.com/orhun/git-cliff-action/blob/main/entrypoint.sh

Reply:
That action can probably be fixed so it doesn't have to do any munching of repository file ownership before execution. That was always an ugly work around in the first place.

Reply:
BTW, if you are using a Docker based runner there isn't much reason to have a separate repo for a GH Action, it works quite well (in spite of the docs) to have it just use the upstream repository *as* the action since it can pull everything prebuilt from GHCR anyway. I'm doing this on quite a number of projects (e.g. `luacheck`, `sile`, etc.).